File: app/src/main/java/com/uramonk/androidtemplateapp/view/NextFragment.java
package com.uramonk.androidtemplateapp.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.trello.rxlifecycle.components.RxFragment;
import com.uramonk.androidtemplateapp.R;
import com.uramonk.androidtemplateapp.databinding.FragmentNextBinding;
import com.uramonk.androidtemplateapp.viewmodel.NextFragmentViewModel;

/**
 * Created by uramonk on 2016/07/04.
 */
public class NextFragment extends RxFragment {
    private FragmentNextBinding binding;

    public static NextFragment newInstance() {
        return new NextFragment();
    }

    public NextFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_next, container, false);
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        binding = FragmentNextBinding.bind(getView());
        NextFragmentViewModel nextFragmentViewModel = new NextFragmentViewModel(this);
        binding.setNextFragmentViewModel(nextFragmentViewModel);
    }
}

File: app/src/main/java/com/uramonk/androidtemplateapp/presentation/view/NextFragment.kt
package com.uramonk.androidtemplateapp.presentation.view

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.trello.rxlifecycle.components.RxFragment
import com.uramonk.androidtemplateapp.R
import com.uramonk.androidtemplateapp.databinding.FragmentNextBinding
import com.uramonk.androidtemplateapp.presentation.viewmodel.NextFragmentViewModel

/**
 * Created by uramonk on 2016/07/04.
 */
class NextFragment : RxFragment() {
    private lateinit var binding: FragmentNextBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_next, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        binding = FragmentNextBinding.bind(view)
        val nextFragmentViewModel = NextFragmentViewModel(this)
        binding.nextFragmentViewModel = nextFragmentViewModel
    }

    companion object {
        fun newInstance(): NextFragment {
            return NextFragment()
        }
    }
}
